scripts/generateCliCode.ts

```diff
@@ -258,16 +265,23 @@ class Generator {
     }
   }
 
-  protected walkProperty(propertyName: string, preference: yamlObject, settingsTree: settingsTreeType): void {
+  protected walkProperty(
+    propertyName: string,
+    preference: yamlObject,
+    notAvailable: boolean,
+    settingsTree: settingsTreeType): void {
```

Comment:
There's enough parameters that I'm starting to wonder if we should pass in an object :) (Don't need to do anything yet, just something to consider for later.)

Reply:
It's only a problem with `updateLeaf`, and I would just put the variants of property names, types, and values in an object, and leave preference`, `notAvailable`, and `settingsTree` as separate parameters.

Reply:
I tested it out. The change adds 44 lines and removes 13.

The three calls to `updateLeaf` call a function to wrap five values in an object.

`updateLeaf` then peels them out of that object.

It doesn't seem worth it to me.

Reply:
Seems reasonable. I was hoping it would make for a better API, but if not then it isn't.